### PR TITLE
GEA-1434 event signature

### PR DESCRIPTION
**Pull request by pangea-andrest**
- Add event signature on log and signature verification on search features
- I left one FIXME comment about options.verify, @ggallien13  let me know what you think about that. After your comment I'll remove FIXME and merge branch

### CODE REVIEW

File: src/services/audit.ts [hunk 1]
```diff
@@ -167,6 +176,12 @@ class AuditService extends BaseService {
       }
     }
 
+    if (options?.verifySignature) {
+      this.verifySignature = options.verifySignature;
+    } else {
```
Comment (ggallien13) [inline src/services/audit.ts:181]: The else condition isn't needed, as this.verifySignature is already initialized to false.

Comment (pangea-andrest) [reply src/services/audit.ts:181]: Yes, but what happen if you reuse AuditService object and change options.verifySignature between calls? I mean, first you call with verifySignature on true, then you make another call with verifySignature in false. Don't you need to set it back to false?

File: src/services/audit.ts [hunk 2]
```diff
@@ -151,6 +158,8 @@ class AuditService extends BaseService {
     // Store the verify mode for the search, used by results
     // Include proofs, root and hash, if verify is set
     if (options?.verify) {
+      // FIXME: If I do a search with verify true, and then another search without verify options. Verify response will remain on true, isn't it?
+      // So I'll try to verify response then and I wont have the information needed.
```
Comment (pangea-andrest) [inline src/services/audit.ts:162]: @ggallien13  it's the same case that here. Don't we have to add the else statement?

File: src/services/audit.ts (hunk 1 above)
Comment (ggallien13) [reply src/services/audit.ts]: Yes, you're correct, the value will remain set to true without the else.

File: src/services/audit.ts (hunk 2 above)
Comment (ggallien13) [reply src/services/audit.ts]: Yes, we need an else here as well to ensure the flag gets reset to false.

File: src/types.ts [hunk 3]
```diff
@@ -30,14 +30,17 @@ export interface ResponseObject<M> {
...
 
   export interface LogData extends Audit.LogOptions {
     event: Audit.Event;
+    signature?: string;
+    public_key?: string;
   }
 
   export interface Event {
-    message: string;
+    message: string | JSON;
```
Comment (ggallien13) [inline src/types.ts:43]: I was wrong about this, it needs to be `string | object`.

**Review by ggallien13 — APPROVED** [review]

**Pull request merged**